e2e/machine_migration_test.go

```diff
@@ -329,6 +396,39 @@ func verifyCAPIMachinePausedCondition(capiMachine *clusterv1.Machine, authority
 	)
 }
 
+func verifyCAPIMachineRemoved(cl client.Client, machineName string) {
+	By(fmt.Sprintf("Verifying the CAPI Machine %s is removed", machineName))
+	Eventually(func() bool {
+		err := komega.Get(&clusterv1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      machineName,
+				Namespace: capiframework.CAPINamespace,
+			},
+		})()
+		return apierrors.IsNotFound(err)
```

Comment:
We could use the capiframework.GetMachine() here

Reply:
Can we do something like:

```go
Eventually(k.Get(capiMachine), timeout).Should(WithTransform(apierrors.IsNotFound, BeTrue())
```